cassdegrees/static/js/staff/rules.js

```diff
@@ -831,10 +831,20 @@ Vue.component('rule_either_or', {
             this.details.either_or[group].splice(index, 1);
             this.do_redraw();
         },
+        duplicate_rule: function(index, group) {
+            // JSON.parse(JSON.stringify(...)) is done to actually duplicate the contents of the rule, rather than just copying the memory references.
+            this.details.either_or[group].splice(index, 0, JSON.parse(JSON.stringify(this.details.either_or[group][index])));
```

Comment:
JSON encoding/decoding is ugly here, but this is JavaScript, so its all ugly anyway :P - does the job anyway.

Reply:
HAHA yeah, I did not like this solution but there was no better way...